cluster-autoscaler/cloudprovider/azure/test/e2e/e2e_suite_test.go

```diff
@@ -53,11 +77,41 @@ var _ = BeforeSuite(func() {
 	vmss, err = armcompute.NewVirtualMachineScaleSetsClient(os.Getenv("AZURE_SUBSCRIPTION_ID"), azCred, nil)
 	Expect(err).NotTo(HaveOccurred())
 
-	k8sConfig := genericclioptions.NewConfigFlags(false)
-	restConfig, err := k8sConfig.ToRESTConfig()
+	restConfig, err := helmEnv.RESTClientGetter().ToRESTConfig()
```

Comment:
Eventually we're going to want to be able to switch between installing cluster-autoscaler on the workload cluster (AKS and Azure self-managed k8s scenarios) or the CAPI mgmt cluster (Cluster API provider scenario). I think the way this is now should allow for that evolution fairly easily, but just FYI.